hello/src/hello.c

```diff
@@ -0,0 +1,18 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "hello.h"
+
+char *
+hello(char *str) {
+  int size = strlen(str) + 10;
+  char *message = (char *) malloc(size);
+
+  if (message == NULL) {
+    exit(EXIT_FAILURE);
```

Comment:
I recommend to return NULL.
NULL means null pointer.
hello() returns (char*) pointer, such function returns null pointer on error normally.

example: fopen() returns NULL when could not open
http://www9.plala.or.jp/sgwr-t/c/sec17.html